File: angular-tailwind-main-CRMFront/src/app/modules/models/user-dto.ts
import { AppRole } from './app-role'; // Si vous avez un modèle de rôle

export interface UserDto {
selected: boolean;
uuid :string;
  id: string;               // ID unique de l'utilisateur
  email: string;            // Email de l'utilisateur
  firstName: string;        // Prénom
  lastName: string;         // Nom
  phone: string;            // Numéro de téléphone
  address: string;          // Adresse
  password: string;         // Mot de passe
  confirmPassword: string;  // Confirmation du mot de passe
  roles: AppRole[];         // Rôles associés à l'utilisateur
  acceptTerms: boolean; 
  enabled: boolean; 
  createdAt:String
    // Accepte les termes et conditions
}

File: RHMANAGER/src/main/java/com/example/User_Service/dto/UserDto.java
package com.example.User_Service.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class UserDto {
    private String uuid;
    private String email;
    private String firstName;
    private String lastName;
    private String phone;
    private String address;
    private LocalDate createdAt;
    private List<RoleDto> roles;
    private Boolean enabled ;

}
